tests/console/yast2_rmt.pm

```diff
@@ -0,0 +1,85 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary : Test for the yast2-rmt module
+# Maintainer Jonathan Rivrain <JRivrain@suse.com>
+
+use base "console_yasttest";
+use strict;
+use utils;
+use testapi;
+
+
+sub password_twice {
+    type_string "$password";
+    send_key "tab";
+    type_string "$password";
+    send_key "alt-o";
+}
+
+sub test_ui {
+    script_run("yast2 rmt; echo yast2-rmt-server-status-\$? > /dev/$serialdev", 0);
+    assert_screen "yast2_rmt_registration";
+    send_key "alt-n";
+    assert_screen "yast2_rmt_ignore_registration_dialog";
+    send_key "alt-i";
+    send_key "ret";
+    assert_screen "yast2_rmt_db_password";
+    send_key "alt-p";
+    type_string "$password";
+    send_key "alt-n";
+    assert_screen "yast2_rmt_db_root_password";
+    password_twice;
+    assert_screen "config_written_successfully";
+    send_key "alt-o";
+    assert_screen "yast2_rmt_ssl";
+    send_key "alt-o";
+    send_key_until_needlematch('yast2_rmt_common_name_empty', 'backspace');
```

Comment:
Did you try if it is reliable to delete 10 characters and check for empty text instead of generating all those screenshots. It would be more clear.

Reply:
I think this way is more reliable in case of config change

Reply:
That thing is slow and ugly, but that's the most reliable we have AFAIK, we use it in several ncurses tests. It would be nice to find something more efficient though.

Reply:
I was talking about the current thing, send_key_until_needlematch. We could do some new function that sends x keys until it matches that we could call this way :  "send_x_keys_until_needlematch( 'needle_name', 'backspace', 10) ". It also looks a bit dirty but would be a lot faster than the current solution.

Reply:
But in the current case I think I'll just do as you say, it looks like good idea.

Reply:
You can do it the "lazy user" way: Just press the key *often enough* so that you can be sure the field is empty, e.g. `for (1 .. 20) { send_key 'backspace'; }` like in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/yast2_samba.pm#L135 . The `assert_screen 'needle_name'` afterwards is even optional then.

Reply:
I just looked at https://github.com/os-autoinst/os-autoinst/blob/7dbadeb6c6b850d05aaa799bff3dcb03484f3c6e/testapi.pm#L1201 and realized it was pointless to use it in this context, since it deletes only 20 characters by default. Then it is almost the same as for (1 .. 20) { send_key 'backspace'; }, but only slower. I'll remove it.